File: Lab9ZastavniyKI305/KI305/Zastavnyi/lab9/Size.py
class Size:
    def __init__(self, width=0, height=0, length=0):
        """
        Конструктор класу Size.

        Ініціалізує атрибути розміру рослини.

        Args:
        - width (float): Ширина рослини.
        - height (float): Висота рослини.
        - length (float): Довжина рослини.

        """
        self.width = width
        self.height = height
        self.length = length

    def get_width(self):
        """
        Отримання ширини рослини.

        Returns:
        - float: Ширина рослини.

        """
        return self.width

    def set_width(self, width):
        """
        Встановлення ширини рослини.

        Args:
        - width (float): Ширина рослини.

        """
        self.width = width

    def get_height(self):
        """
        Отримання висоти рослини.

        Returns:
        - float: Висота рослини.

        """
        return self.height

    def set_height(self, height):
        """
        Встановлення висоти рослини.

        Args:
        - height (float): Висота рослини.

        """
        self.height = height

    def get_length(self):
        """
        Отримання довжини рослини.

        Returns:
        - float: Довжина рослини.

        """
        return self.length

    def set_length(self, length):
        """
        Встановлення довжини рослини.

        Args:
        - length (float): Довжина рослини.

        """
        self.length = length

    def get_status(self, fout):
        """
        Виведення статусу розміру рослини.

        Args:
        - fout (File): Файловий об'єкт для запису.

        """
        print("Ширина:", self.width)
        print("Довжина:", self.length)
        print("Висота:", self.height)

        fout.write("Ширина: " + str(self.width) + "\n")
        fout.write("Довжина: " + str(self.length) + "\n")
        fout.write("Висота: " + str(self.height) + "\n")

    def change_size_to_grow(self):
        """
        Збільшення розміру рослини.

        """
        self.set_width(self.width + 1.1)
        self.set_height(self.height + 1.1)
        self.set_length(self.length + 1.1)

    def change_size_to_death(self):
        """
        Зміна розміру рослини на нуль (для "смерті").

        """
        self.set_width(0)
        self.set_height(0)
        self.set_length(0)

File: Lab3ZastavnyiKI305/Lab3ZastavnyiKI305/src/KI305/Zastavnyi/Lab3/Size.java
package KI305.Zastavnyi.Lab3;

import java.io.PrintWriter;

import static java.lang.System.out;

/**
 * Клас, що представляє розміри об'єкта. Кожен розмір може мати ширину, висоту та довжину.
 * Для спрощення можливості створення об'єктів за замовчуванням та з заданими параметрами є конструктори.
 */
public class Size {
    private double width;   // Ширина
    private double height;  // Висота
    private double length;  // Довжина

    /**
     * Конструктор за замовчуванням. Встановлює всі властивості розміру на значення 0.
     */
    public Size() {
        this.width = 0;
        this.height = 0;
        this.length = 0;
    }

    /**
     * Конструктор, який дозволяє встановити ширину, висоту та довжину розміру при створенні об'єкта.
     *
     * @param width  Ширина
     * @param height Висота
     * @param length Довжина
     */
    public Size(double width, double height, double length) {
        this.width = width;
        this.height = height;
        this.length = length;
    }

    /**
     * Метод, що повертає ширину розміру.
     *
     * @return Ширина
     */
    public double getWidth() {
        return width;
    }

    /**
     * Метод, що встановлює ширину розміру.
     *
     * @param width Ширина
     */
    public void setWidth(double width) {
        this.width = width;
    }

    /**
     * Метод, що повертає висоту розміру.
     *
     * @return Висота
     */
    public double getHeight() {
        return height;
    }

    /**
     * Метод, що встановлює висоту розміру.
     *
     * @param height Висота
     */
    public void setHeight(double height) {
        this.height = height;
    }

    /**
     * Метод, що повертає довжину розміру.
     *
     * @return Довжина
     */
    public double getLength() {
        return length;
    }

    /**
     * Метод, що встановлює довжину розміру.
     *
     * @param length Довжина
     */
    public void setLength(double length) {
        this.length = length;
    }

    /**
     * Метод, що виводить статус розміру на консоль та у файл за допомогою PrintWriter.
     *
     * @param fout PrintWriter об'єкт для запису у файл
     */
    public void getStatus(PrintWriter fout) {
        out.println("Ширина: " + this.width);
        out.println("Довжина: " + this.length);
        out.println("Висота: " + this.height);

        fout.println("Ширина: " + this.width);
        fout.println("Довжина: " + this.length);
        fout.println("Висота: " + this.height);
    }

    /**
     * Метод, що змінює розмір на зростання.
     */
    public void changeSizeToGrow() {
        setWidth(this.width + 1.1);
        setHeight(this.height + 1.1);
        setLength(this.length + 1.1);
    }

    /**
     * Метод, що змінює розмір на "Смерть", встановлюючи всі властивості розміру на значення 0.
     */
    public void changeSizeToDeath() {
        setWidth(0);
        setHeight(0);
        setLength(0);
    }
}
